Duplicati/Library/Utility/ReadLimitLengthStream.cs

```diff
@@ -35,12 +66,12 @@ public override void Flush()
             // NOOP
         }
 
-        public override long Length => m_endPosition;
+        public override long Length => m_length;
```

Comment:
This appears to change the behavior, but your version makes more sense to me.  Was the previous implementation a bug, or was this property not read?

Reply:
If you're referring to the Length property, I suppose you are right that if the innerStream wasn't starting at 0, then this length wouldn't have been correct. But this class was only used with new streams, so the bug likely didn't ever come up.